streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -46,6 +48,16 @@
 import java.util.NoSuchElementException;
 import java.util.Set;
 
+/**
+ * A persistent key-value store based on RocksDB.
+ *
+ * Note that array typed keys may result in incorrect caching behavior.
```

Comment:
Perhaps rewrite this doc snippet to:

> Note that the use of array-typed keys is discouraged because they result in incorrect behavior for ordering.  If you are working on byte arrays as key, for example, you may want to wrap them with the `Bytes` class, i.e. use `RocksDBStore<Bytes, ...>` rather than `RocksDBStore<byte[], ...>`.

(Same for the other docstrings that include this note.)